internal/configmodels/configmodels.go

```diff
@@ -0,0 +1,157 @@
+type ConfigV2 struct {
+	Receivers  []ReceiverCfg  `mapstructure:"-"`
+	Exporters  []ExporterCfg  `mapstructure:"-"`
+	Processors []ProcessorCfg `mapstructure:"-"`
+	Pipelines  Pipelines      `mapstructure:"pipelines"`
+}
+
+// ReceiversKeyName is the configuration key name for receivers section.
+const ReceiversKeyName = "receivers"
+
+// ExportersKeyName is the configuration key name for exporters section.
+const ExportersKeyName = "exporters"
+
+// ProcessorsKeyName is the configuration key name for processors section.
+const ProcessorsKeyName = "processors"
+
+// Pipelines defines the pipelines section of configuration.
+type Pipelines struct {
+	Traces  []*Pipeline `mapstructure:"traces"`
+	Metrics []*Pipeline `mapstructure:"metrics"`
+}
+
+// Pipeline defines a single pipeline.
+type Pipeline struct {
+	Name       string              `mapstructure:"name"`
+	Receivers  []PipelineReceiver  `mapstructure:"receivers"`
+	Processors []PipelineProcessor `mapstructure:"ordered-processors"`
+	Exporters  []PipelineExporter  `mapstructure:"exporters"`
+}
+
+// PipelineReceiver defines a receiver of a pipeline.
+type PipelineReceiver struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineProcessor defines a processor of a pipeline.
+type PipelineProcessor struct {
+	Name string `mapstructure:"name"`
+}
+
+// PipelineExporter defines an exporter of a pipeline.
+type PipelineExporter struct {
+	Name string `mapstructure:"name"`
+}
+
+// NamedEntity defines a configuration entity that has a name.
+type NamedEntity interface {
+	GetName() string
+}
+
+// ReceiverCfg is the configuration of a receiver. Specific receivers
+// must implement this interface and will typically embed ReceiverCommon
+// struct or a struct that extends it.
+type ReceiverCfg interface {
+	NamedEntity
+}
+
+// TypeKeyName must match Type field key names for Receiver and Exporter.
+const TypeKeyName = "type"
+
+// ExporterCfg is the configuration of an exporter. Specific exporters
+// must implement this interface and will typically embed ExporterCommon
+// struct or a struct that extends it.
+type ExporterCfg interface {
+	NamedEntity
+}
+
+// ProcessorCfg is the configuration of a processor. Specific processors
+// must implement this interface and will typically embed ProcessorCommon
+// struct or a struct that extends it.
+type ProcessorCfg interface {
+	NamedEntity
+}
+
+// Below are common setting structs for Receivers, Exporters and Processors.
+// These are helper structs which you can use to implement your specific
+// receiver/exporter/processor config storage.
+
+// ReceiverCommon defines common settings for a single-protocol receiver configuration.
+// Specific receivers can embed this struct and extend it with more fields if needed.
+type ReceiverCommon struct {
+	Type     string `mapstructure:"type"`
+	Enabled  bool   `mapstructure:"enabled"`
+	Name     string `mapstructure:"name"`
+	HostPort string `mapstructure:"hostport"`
```

Comment:
I think my preference is Endpoint, because if you use a DNS address that can be also used here, unless you have a clear example in mind where that is not ok.

Reply:
Endpoint sounds better than HostPort. I will change.